### ISSUE TITLE
No download with 1.0.0

### ISSUE THREAD

**Issue by Roflkopter666** [althonos/InstaLooter#157]
## Library version

*What's the installed library version ? Check with `instalooter --version`*:

```
instalooter v1.0.0
```

## Environment

*Describe here your environment, including:*

* Arch Linux
* Python 3.6.4, 2.7.14

## Error description - runtime

No error, instalooter is stuck.

### Reproducible test case

```
~/.local/bin/instalooter hashtag funny funny  -T {username}.{datetime}.{id} -N
```

### Expected behaviour

Download images

### Actual behaviour

2018-04-06 19:02:08 workstation instalooter.cli[3467] NOTICE Starting download of `funny`
Now nothing happens.
I traced it down to pages.py line 76 which in turn executes the while loop starting at line 57 endlessly.

**Comment by Philosophy11** [comment]
Hey, I'm seeing what seems like the same issue. Using instaLooter v1.0.0 on Windows, python 3.6.5. Using as an API in code. Worked fine a few days ago but now hangs forever. Here's my API test case:

from instalooter.looters import ProfileLooter
looter=ProfileLooter("missjazminad")
looter.download(r"D:\TestFolder", media_count=10)

Code is able to create looter var, but pauses on download step. Never completes or give an error. I did some testing and it seems to pause in looters.py in a few places. First at ln 424 (dlpbar creation, which I'm not using). If I comment those lines out, next at ln 583 (while in _fill_media_queue).

**Comment by anongit** [comment]
> I traced it down to pages.py line 76 which in turn executes the while loop starting at line 57 endlessly.

Same problem, I added `print(data)` there to see the error:
```
{"message": "forbidden", "status": "fail"}
```

Adding the `ig_pr=1` cookie helps:
```python
 with session.get(self._geturl(self._cursor), cookies={'ig_pr': '1'}) as res:
    data = res.json()
```
I think they set in `Consumer.js` script, at least it's mentioned there.

**Comment by sanjeevmagoo** [comment]
Any workaround or fix?

**Comment by AbysmalBiscuit** [comment]
Guys, this is an issue with instagram removing the api call that returns a nice json response that instalooter uses to get the media download links.

To fix this a rewrite of the pages that instalooter fetches from instagram and the way it extracts the download links from the pages is necessary. 

Quoting @althonos from #159:
>I'll try to look into this, looks like the API that allowed to get JSON directly from a public page is not available anymore. I'll have to resign to using a regex as I did before.

**Comment by althonos** [comment]
Unfortunately that's not as simple, since I was also using *graphql* queries which format has internally changed. So I can still extract some of the public data I need (like, from a profile page) but for now I cannot use *graphql* to iterate through the posts.

I just have to figure out what are the needed headers and cookies for the queries to start working again.

**Comment by althonos** [comment]
@anongit : I got it to work in the previous commit, thanks.

**Comment by althonos** [comment]
Should be fixed in `v2.0.0`. Take note that since rate limits were drastically decreased, you'll not be able to download a lot of medias before being shut out. I advise logging in as the amount of authorized queries is a lot higher when using a logged in session.
